<?xml version="1.0"?>
<!--
Copyright (c) 2018, STEREOLABS.

All rights reserved.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-->
<launch>
    <arg name="svo_file" default=""/>

    <!-- Coordinate frame -->
    <arg name="camera_model"         default="0" /> <!-- 0=ZED, 1=ZEDM-->
    <arg name="serial_number"        default="17684" />
    <arg name="verbose"              default="true" />
    <arg name="gpu_id"              default="-1" />
    <arg name="resolution"           default="2" /> <!--0=RESOLUTION_HD2K, 1=RESOLUTION_HD1080, 2=RESOLUTION_HD720, 3=RESOLUTION_VGA -->
    <arg name="frame_rate"           default="60" />
    <!-- RESOLUTION_HD2K   -> 2208*1242, available framerates: 15 fps.
         RESOLUTION_HD1080 -> 1920*1080, available framerates: 15, 30 fps.
         RESOLUTION_HD720  -> 1280*720,  available framerates: 15, 30, 60 fps.
         RESOLUTION_VGA    -> 672*376,   available framerates: 15, 30, 60, 100 fps. -->

    
    <node pkg="lims2_vision" type="bullet_track_zed" name="bullet_track_zed" output="screen">     
        <!-- SVO file path -->
        <param name="svo_filepath"          value="$(arg svo_file)" />

        <!-- ZED parameters -->
        <param name="serial_number"         value="$(arg serial_number)" />

        <param name="verbose"               value="$(arg verbose)" />
        <param name="quality"               value="1" />
        <param name="sensing_mode"          value="0" />
        <param name="gain"                  value="100" />
        <param name="exposure"              value="100" />
        <param name="auto_exposure"         value="true" />
    </node>
</launch>
